src/Exception/RequestException.php

```diff
@@ -102,13 +105,15 @@ public static function create(
             $response->getReasonPhrase()
         );
 
-        $summary = \GuzzleHttp\Psr7\Message::bodySummary($response);
+        $summary = $bodySummarizer === null
+            ? \GuzzleHttp\Psr7\Message::bodySummary($response)
+            : $bodySummarizer($response);
```

Comment:
PHP-http has the concept of "message formatter" which would be ideal here. 

Im not a bug fan of using a callable without a technical contract (eg interface).

Reply:
I don't think Guzzle depends on php-http, intentionally so?

Reply:
It doesn’t and it shouldn’t. But I like the concept. We could add a similar interface to guzzle.

Reply:
I mean, given that it is `callable(MessageInterface): string` and not just `callable`, I think it's fine, and an interface would be overkill, but I can add it if you really want.

Reply:
Yeah. I believe that would be a good idea.

Reply:
We already have a different `MessageFormatterInterface`. I guess I should name this new one `BodySummarizerInterface`?

Reply:
Yeah. Sounds good